Fix Picker layout on Mac Catalyst 26+

kubaflo opened the pull request:
On Mac Catalyst 26+, the Picker view was incorrectly positioned, appearing visually shifted within the picker controller.
This PR centers UIPickerView using NSLayoutConstraint when added to the pickerController’s view. This ensures correct alignment, predictable sizing, and consistent layout behavior across Mac Catalyst versions.

Fixes https://github.com/dotnet/maui/issues/33229

|Before|After|
|--|--|
|<img src="https://github.com/user-attachments/assets/d55949ec-3f78-4836-a18c-778e240f2a1f" width="300px"/>|<img src="https://github.com/user-attachments/assets/7fb709ba-b1cd-490e-b093-7dd79f48ea08" width="300px"/>|

kubaflo committed:
Add layout constraints to PickerView on iOS

PickerView is now centered and sized using NSLayoutConstraint when added to the pickerController's view. This ensures proper alignment and sizing, improving layout consistency.

PureWeen reviewed (changes requested):
Build issue: `pickerController` on line 87 should be `_pickerController` (missing underscore). There is no local variable named `pickerController` in this method — only the field `_pickerController`. This will fail to compile on MacCatalyst.

Minor suggestion: Since the new constraints fully define the picker's layout, the initial `frame` passed to `new UIPickerView(frame)` on line 53 is now effectively ignored. Consider simplifying to `new UIPickerView()` to make it clear constraints drive the layout.

kubaflo committed:
Remove explicit frame use for UIPickerView

Remove the unused RectangleF (CGRect) using and eliminate the explicit frame/RectangleF variable for UIPickerView in PickerHandler.iOS. The code now instantiates UIPickerView with the default constructor, then sets its Model and reloads components. This simplifies the code and removes the unnecessary CoreGraphics alias and frame creation.
